Storage/SQL/SQLiteHistoryRecommendations.swift

```diff
@@ -0,0 +1,64 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+import Foundation
+import Shared
+import XCGLogger
+import Deferred
+
+extension SQLiteHistory: HistoryRecommendations {
+    public func getHighlights() -> Deferred<Maybe<Cursor<Site>>> {
+        let limit = 20
+        let microsecondsPerMinute: Timestamp = 60_000_000 // 1000 * 1000 * 60
+        let now = NSDate.nowMicroseconds()
+        let thirtyMinutesAgo = now - 30 * microsecondsPerMinute
+        let threeDaysAgo = now - (60 * microsecondsPerMinute) * 24 * 3
+        let bookmarkLimit = 1
+        let historyLimit = limit - bookmarkLimit
+
+        let highlightProjection = "historyID, url, title, guid, visitCount, visitDate, iconID, iconURL, iconType, iconDate, iconWidth"
+
+        let nonRecentHistory = [
+            "SELECT * FROM (",
+            "   SELECT \(TableHistory).id as historyID, url, title, guid, visitDate, (SELECT COUNT(1) FROM \(TableVisits) WHERE s = \(TableVisits).siteID) AS visitCount",
+            "   FROM (",
+            "       SELECT siteID AS s, max(date) AS visitDate",
+            "       FROM \(TableVisits)",
+            "       WHERE date < \(thirtyMinutesAgo)",
+            "       GROUP BY siteID",
+            "   )",
+            "   LEFT JOIN \(TableHistory) ON \(TableHistory).id = s",
+            "   WHERE visitCount <= 3 AND title NOT NULL AND title != ''",
+            "   LIMIT \(historyLimit)",
+            ")"
+        ].joinWithSeparator(" ")
```

Comment:
Implied: do this with string concatenation instead of dynamically using `joinWithSeparator`. I think the Swift compiler is much better at type analysis now, so this shouldn't blow up compile time, but probably not smart enough to recognize `joinWithSeparator` on constants.